url/src/lib.rs

```diff
@@ -1742,6 +1742,37 @@ impl Url {
         let old_after_path_pos = to_u32(self.serialization.len()).unwrap();
         let cannot_be_a_base = self.cannot_be_a_base();
         let scheme_type = SchemeType::from(self.scheme());
+        let mut path_empty = false;
+
+        // Check ':' and then see if the next character is '/'
+        let mut has_host = if let Some(index) = self.serialization.find(":/") {
```

Comment:
Actually, can we change this to - self.serialization.find(":") then check if the next char is `/` ?
That way we don't have to call find twice.